kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/internal/parquet/TestParquetBatchReader.java

```diff
@@ -540,4 +556,38 @@ private static Tuple2<ColumnarBatch, Integer> getBatchForRowId(
 
         throw new IllegalArgumentException("row id is not found: " + rowId);
     }
+
+    private static <T> void checkArrayValue(
+            ArrayValue arrayValue, DataType expDataType, List<T> expList) {
+        int size = expList.size();
+        ColumnVector elementVector = arrayValue.getElements();
+        // Check the size is as expected and arrayValue.getSize == elementVector.getSize
+        assertEquals(size, arrayValue.getSize());
+        assertEquals(size, elementVector.getSize());
+        // Check the element vector has the correct data type
+        assertEquals(elementVector.getDataType(), expDataType);
+        // Check the elements are correct
+        assertEquals(expList, VectorUtils.toJavaList(arrayValue));
+        // TODO check that you cannot access later elements (can we upgrade JUnit ot 4.13?)
```

Comment:
@vkorukanti any reason we can't do this upgrade?

Reply:
I don't have any reason. If needed we can upgrade given this is just for test.